EMentors/app/controllers/application_controller.rb

```diff
@@ -2,12 +2,19 @@ class ApplicationController < ActionController::Base
     before_action :configure_permitted_parameters, if: :devise_controller?
     protected
     def after_sign_in_path_for(resource)
-        home_index_path
+        if current_user.role == 'teacher'
```

Comment:
please add a method is_teacher? and directly call 
`if resource.is_teacher? ? teachers_index_path : students_index_path`